.github/hooks/pre-commit

```diff
@@ -0,0 +1,17 @@
+#!/bin/sh
+
+#
+# Install this using:
+# ln -s ../../.github/hooks/pre-commit .git/hooks/pre-commit
+#
+
+FILES=$(git diff --cached --name-only --diff-filter=ACMR "*.dart" | sed 's| |\\ |g')
+[ -z "$FILES" ] && exit 0
+
+# Prettify all selected files
+echo "$FILES" | xargs dart format -l 120
```

Comment:
What should happen if it is not in $PATH? Also how do you do that? Could you just change this to your liking?

Reply:
The hook should abort with an error explaining the issue in my opinion. Here is how to do it: https://stackoverflow.com/a/677212/1763110

We also recommend to use `fvm`. In this case the dart command does not need to be in the path. Instead you use `fvm dart`.

So actually this script should use fvm instead of `dart` directly.

Reply:
Done. Sadly this does not work for me when committing from IntelliJ as the path env is different and does not include fvm for me. Dart and flutter are found though :/ Seems to be this issue: https://youtrack.jetbrains.com/issue/IDEA-220910
Will probably just keep a modified version of the hook on my machine.

Reply:
Yes fvm is probably not in the path of intellij with a default installation. This can be changed by adding fvm to the path in ~/.xprofile file usually.